src/components/Meals.tsx

```diff
@@ -1,66 +1,65 @@
 import React, { useState } from 'react';
-import { View, Text, TextInput, TouchableOpacity, Button } from 'react-native';
+import { View, Text, TextInput, TouchableOpacity } from 'react-native';
 import styles from '../styles';
 import { MealCategories } from '../models/meals.model';
 
 type MealsComponentProps = {
   mealCategories: MealCategories[];
-  handleBreakfastText: (text: string, category: MealCategories) => void;
+  handleMealText: (text: MealText, category: MealCategories) => void;
 };
 
-const Meals: React.FC<MealsComponentProps> = ({ mealCategories, handleBreakfastText }) => {
-  const [textMeal, setMealText] = useState('');
-  const [breakfastText, setBreakfastText] = useState('');
-  const [lunchText, setLunchText] = useState('');
-  const [dinnerText, setDinnerText] = useState('');
+type MealText = string;
 
-  const onMealEnter = (text: string, category: MealCategories) => {
-    switch (category) {
-      case MealCategories.BREAKFAST:
-        setBreakfastText(text);
-        break;
-      case MealCategories.LUNCH:
-        setLunchText(text);
-        break;
-      case MealCategories.DINNER:
-        setDinnerText(text);
-        break;
-      default:
-        break;
-    }
+const Meals: React.FC<MealsComponentProps> = ({ mealCategories, handleMealText }) => {
+  const [mealText, setMealText] = useState<Record<MealCategories, MealText>>({
```

Comment:
why do we need the word `Text` in the name? isn't that a little too abstract? (same for `showEnteredText`)
what is this exactly a state of?

Reply:
Sounds better